compose/config.py

```diff
@@ -311,6 +314,18 @@ def resolve_host_path(volume, working_dir):
         return container_path
 
 
+def resolve_build_path(build_path, working_dir=None):
+    if working_dir is None:
+        raise Exception("No working_dir passed to resolve_build_path")
+
+    _path = os.path.join(working_dir, build_path)
+    _path = os.path.normpath(os.path.abspath(_path))
+    if (not os.path.exists(_path) or not os.access(_path, os.R_OK)):
```

Comment:
You can use the `expand_path` helper here, like `get_env_files` and `resolve_host_path` do.